sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -104,6 +118,115 @@ class SentryOkHttpInterceptor(
         }
     }
 
+    // TODO: ignore exceptions of the type UnknownHostException
+
+    private fun captureEvent(request: Request, response: Response) {
+        // not possible to get a parameterized url, but we remove at least the
+        // query string and the fragment.
+        // url example: https://api.github.com/users/getsentry/repos/#fragment?query=query
+        // url will be: https://api.github.com/users/getsentry/repos/
+        // ideally we'd like a parameterized url: https://api.github.com/users/{user}/repos/
+        // but that's not possible
+        var requestUrl = request.url.toString()
+
+        val query = request.url.query
+        if (!query.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("?$query", "")
+        }
+
+        val urlFragment = request.url.fragment
+        if (!urlFragment.isNullOrEmpty()) {
+            requestUrl = requestUrl.replace("#$urlFragment", "")
+        }
+
+        if (!captureFailedRequests || !TracePropagationTargets.contain(failedRequestsTargets, requestUrl) || !containsStatusCode(response.code)) {
+            return
+        }
+
+        val mechanism = Mechanism().apply {
+            type = "SentryOkHttpInterceptor"
+        }
+        val exception = SentryHttpClientError("Event was captured because the request status code was ${response.code}")
+        val mechanismException = ExceptionMechanismException(mechanism, exception, Thread.currentThread(), true)
+        val event = SentryEvent(mechanismException)
+
+        val hint = Hint()
+        hint.set("request", request)
+        hint.set("response", response)
+
+        // TODO: remove after fields indexed
+        val tags = mutableMapOf<String, String>()
+        tags["status_code"] = response.code.toString()
+        tags["url"] = requestUrl
+
+        val unknownRequestFields = mutableMapOf<String, Any>()
+
+        val sentryRequest = io.sentry.protocol.Request().apply {
+            url = requestUrl
+            // Cookie is only sent if isSendDefaultPii is enabled
+            cookies = if (hub.options.isSendDefaultPii) request.headers["Cookie"] else null
+            method = request.method
+            queryString = query
+            headers = getHeaders(request.headers)
+            fragment = urlFragment
+
+            request.body?.contentLength().ifHasValidLength {
+                // TODO: should be mapped in relay and added to the protocol
+                unknownRequestFields["body_size"] = it
```

Comment:
@jjbayer forgot to mention this one.
`event.request.body_size` is currently being dropped on Relay because it doesn't retain any fields.